core/common/test/files/SmokeFileTest.kt

```diff
@@ -443,6 +443,36 @@ class SmokeFileTest {
         source.close()  // there should be no error
     }
 
+    @Test
+    fun listDirectory() {
+        assertFailsWith<FileNotFoundException> { SystemFileSystem.list(createTempPath()) }
+
+        val tmpFile = createTempPath().also {
+            SystemFileSystem.sink(it).close()
+        }
+        assertFailsWith<IOException> { SystemFileSystem.list(tmpFile) }
+
+        val dir = createTempPath().also {
+            SystemFileSystem.createDirectories(it)
+        }
+        assertEquals(emptyList(), SystemFileSystem.list(dir))
+
+        val subdir = Path(dir, "subdir").also {
+            SystemFileSystem.createDirectories(it)
+            SystemFileSystem.sink(Path(it, "file")).close()
+        }
+        assertEquals(listOf(subdir), SystemFileSystem.list(dir))
+
+        val file = Path(dir, "file").also {
+            SystemFileSystem.sink(it).close()
```

Comment:
Why not test other file types, such as symlinks?

Reply:
We don't support symlinks yet (in wasmWasi it was supported only to make other things work correctly).

Reply:
Symlinks can't be listed from directory?

Reply:
We don't have an API to create them for testing purposes. They can be listed, of course.

Reply:
Issue https://github.com/Kotlin/kotlinx-io/issues/241